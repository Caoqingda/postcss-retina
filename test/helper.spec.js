const {
    getPseudoSelector,
    createMediaQuery,
    createPseudoElement,
    unique
} = require('../lib/helper');

it('getPseudoSelector()', () => {
    expect(getPseudoSelector(['class1', 'class2', 'class3'])).toEqual([1, 2, 3]);// .toMatchSnapshot();
});

it('createMediaQuery()', () => {
    expect(createMediaQuery(['class1', 'class2', 'class3'], 3))
        .toMatchSnapshot();
});

it('createMediaQuery(empty)', () => {
    expect(createMediaQuery([], 3))
        .toMatchSnapshot();
});

it('createPseudoElement()', () => {
    expect(createPseudoElement(['class1', 'class2', 'class3']))
        .toMatchSnapshot();
});

it('createPseudoElement(empty)', () => {
    expect(createPseudoElement([]))
        .toMatchSnapshot();
});

it('unique()', () => {
    expect(unique(['class1', 'class2', 'class1', 'class3']))
        .toMatchSnapshot();
});
