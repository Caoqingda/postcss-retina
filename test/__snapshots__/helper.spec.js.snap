// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`createMediaQuery() 1`] = `
"
/* dpr is 3 */
@media only screen and (-webkit-min-device-pixel-ratio: 3), 
only screen and (min-device-pixel-ratio: 3) {
  class1::before, class2::before, class3::before {
    transform: scale(0.333333);
    transform-origin: 0 0;
    width: 300%;
    height: 300%;
    font-size: 300px;
  }
}"
`;

exports[`createMediaQuery(empty) 1`] = `""`;

exports[`createPseudoElement() 1`] = `
"
class1, class2, class3 {
  position: relative;
}
 
class1::before, class2::before, class3::before {
  content: ' ';
  position: absolute;
  width: 100%;
  height: 100%;
  top: 0;
  left: 0;
  pointer-events: none;
  box-sizing: border-box;
  font-size: 100px;
}"
`;

exports[`createPseudoElement(empty) 1`] = `""`;

exports[`getPseudoSelector() 1`] = `"class1::before, class2::before, class3::before"`;

exports[`unique() 1`] = `
Array [
  "class1",
  "class2",
  "class3",
]
`;
